search/v2/features_filter.cpp

```diff
@@ -23,5 +25,52 @@ unique_ptr<coding::CompressedBitVector> FeaturesFilter::Filter(
     return make_unique<coding::SparseCBV>();
   return coding::CompressedBitVector::Intersect(*m_filter, cbv);
 }
+
+void CBVPtr::Free()
+{
+  if (m_isOwner)
+    delete m_ptr;
+
+  m_ptr = nullptr;
+  m_isOwner = false;
+  m_isFull = false;
+}
+
+void CBVPtr::Union(coding::CompressedBitVector const * p)
+{
+  if (!p || m_isFull)
+    return;
+
+  if (!m_ptr)
+  {
+    m_ptr = p;
+    m_isFull = false;
+  }
+  else
+    Set(coding::CompressedBitVector::Union(*m_ptr, *p).release(), true);
+}
+
+void CBVPtr::Intersect(coding::CompressedBitVector const * p)
```

Comment:
This method is always called for non-null pointers, please, change the argument type to const reference.